src/sonic-frr/dplane_fpm_sonic/dplane_fpm_sonic.c

```diff
@@ -1348,7 +1354,7 @@ static ssize_t netlink_srv6_msg_encode(int cmd,
 		if (!netlink_srv6_localsid_msg_encode(
 				cmd, ctx, data, datalen, fpm, force_nhg))
 			return 0;
-	} else if (!sid_zero(&nexthop->nh_srv6->seg6_segs)) {
+	} else if (!sid_zero((const struct seg6_seg_stack *)nexthop->nh_srv6->seg6_segs)) {
```

Comment:
why the (const struct seg6_seg_stack *) it's already that right?

Reply:
struct nexthop_srv6 {
	/* SRv6 localsid info for Endpoint-behaviour */
	enum seg6local_action_t seg6local_action;
	struct seg6local_context seg6local_ctx;

	/* SRv6 Headend-behaviour */
	struct seg6_seg_stack *seg6_segs;
};
As we can see, nh_srv6->seg6_segs is already a pointer. Putting & was making it a double pointer. So, we removed &. That was the change.
I added (const struct seg6_seg_stack *)  because it was giving a warning.

Reply:
The main problem was the double pointer issue (else if (!sid_zero(&nexthop->nh_srv6->seg6_segs)) {) which I fixed in above code.

Reply:
Done, I have removed the typecasting.